src/Queue/Backend/Beanstalkd/BeanstalkdQueueStoreAdapter.php

```diff
@@ -112,16 +112,18 @@ public function ack(MailJobInterface $mailJob)
         }
 
         $pheanstalk = $this->getConnection()->getInstance()->useTube($this->queueName);
-        if ($mailJob->isCompleted()) {
-            $pheanstalk->delete($mailJob->getPheanstalkJob());
-            return null;
-        }
 
-        $timestamp = $mailJob->getTimeToSend();
-        $delay = max(0, $timestamp - time());
+        $pheanstalk->delete($mailJob->getPheanstalkJob());
```

Comment:
pheanstalk has a method to re-queue the job, the release as it was set.
set the algorithm to be like:

* if not finished, increase the attempt number;
* if not overtaking the number of attempts defined, use release method;
* if it's finished or reached the attempts limit, delete